pair/src/lib.rs

```diff
@@ -0,0 +1,431 @@
+#![no_std]
+
+use gear_lib::fungible_token::{ft_core::*, state::*};
+use gear_lib_derive::{FTCore, FTMetaState, FTStateKeeper};
+use gstd::{cmp, exec, msg, prelude::*, ActorId};
+use num::integer::Roots;
+use pair_io::*;
+pub mod math;
+pub mod messages;
+
+const MINIMUM_LIQUIDITY: u128 = 1000;
+static ZERO_ID: ActorId = ActorId::zero();
+
+#[derive(Debug, Default, FTStateKeeper, FTCore, FTMetaState)]
+pub struct Pair {
+    #[FTStateField]
+    pub token: FTState,
+    pub factory: ActorId,
+    pub token0: ActorId,
+    pub token1: ActorId,
+    last_block_ts: u128,
+    pub balance0: u128,
+    pub balance1: u128,
+    reserve0: u128,
+    reserve1: u128,
+    pub price0_cl: u128,
+    pub price1_cl: u128,
+    pub k_last: u128,
+}
+
+static mut PAIR: Option<Pair> = None;
+
+impl Pair {
+    // INTERNAL METHODS
+    fn _mint(&mut self, to: ActorId) -> u128 {
+        let amount0 = self.balance0.overflowing_sub(self.reserve0).0;
+        let amount1 = self.balance1.overflowing_sub(self.reserve1).0;
+        let fee_on = self._mint_fee(self.reserve0, self.reserve1);
+        let total_supply = self.get().total_supply;
+        let liquidity: u128;
+        if total_supply == 0 {
+            // Math.sqrt(amount0.mul(amount1)).sub(MINIMUM_LIQUIDITY);
+            liquidity = amount0
+                .overflowing_mul(amount1)
+                .0
+                .sqrt()
+                .overflowing_sub(MINIMUM_LIQUIDITY)
+                .0;
+            // add this later to ft lib
+            FTCore::mint(self, &ZERO_ID, liquidity);
```

Comment:
Mint to `zero` address?

Reply:
To permanently lock minimum liquidity tokens. I guess it's not applicable here, since the total_supply is not checked anywhere.